File: lib/domain/usecase/call_recording_usecases.dart
import 'package:support_call_recorder/domain/model/call_recording.dart';
import 'package:support_call_recorder/domain/model/support_status.dart';
import 'package:support_call_recorder/domain/repository/call_recording_repository.dart';

class CallRecordingUseCases {
  const CallRecordingUseCases(this._repository);
  final CallRecordingRepository _repository;

  Future<List<CallRecording>> list() => _repository.getAll();
  Future<CallRecording?> byId(int id) => _repository.getById(id);
  Future<int> save(CallRecording recording) => _repository.insert(recording);
  Future<void> updateNote(int id, String note) =>
      _repository.updateNote(id, note);
  Future<void> updateStatus(int id, SupportStatus status) =>
      _repository.updateStatus(id, status);
}

File: app/src/main/java/com/reco/support/domain/usecase/CallRecordingUseCases.kt
package com.reco.support.domain.usecase

import com.reco.support.domain.model.CallRecording
import com.reco.support.domain.model.SupportStatus
import com.reco.support.domain.repository.CallRecordingRepository
import kotlinx.coroutines.flow.Flow

class ObserveRecordingsUseCase(private val repository: CallRecordingRepository) {
    operator fun invoke(): Flow<List<CallRecording>> = repository.observeAll()
}

class GetRecordingUseCase(private val repository: CallRecordingRepository) {
    suspend operator fun invoke(id: Long): CallRecording? = repository.getById(id)
}

class SaveRecordingUseCase(private val repository: CallRecordingRepository) {
    suspend operator fun invoke(recording: CallRecording): Long = repository.insert(recording)
}

class UpdateRecordingNoteUseCase(private val repository: CallRecordingRepository) {
    suspend operator fun invoke(id: Long, note: String) = repository.updateNote(id, note)
}

class UpdateRecordingStatusUseCase(private val repository: CallRecordingRepository) {
    suspend operator fun invoke(id: Long, status: SupportStatus) = repository.updateStatus(id, status)
}
